bin/blocks-plot

```diff
@@ -0,0 +1,256 @@
+        
+
+    if args.list:
+        # Nicely print all channels contained in the specified experiments
+        print()
+        print("Loaded {} experiment(s):".format(n_experiments))
+        for i, fname in enumerate(experiments):
+            print("    {}: {}".format(i, fname))
+        print()
+        print("Containing the following channels:")
+        for ch in sorted(all_channels):
+            # create a string indicating which experiments contain which 
+            #  channels
+            indicator = []
+            for i, channels in enumerate(channels_per_experiment.values()):
+                if ch in channels:
+                    indicator.append(str(i))
+                else:
+                    indicator.append(" ")
+            indicator = ",".join(indicator)
+            print("    {}: {}".format(indicator, ch))
+        print()
+        return
+
+    if args.channels:
+        channel_specs = args.channels.split(',')
+        matched = match_channel_specs(experiments, channel_specs)
+
+        if args.ipython:
+            from IPython import embed
+            print(IPYTHON_HEADER)
+            embed()
+            return
+
+        plot_dataframe(matched)
+        return
+    else:
+        # Interactive mode
+        def completer(text, state):
+            """Completion callback function for readline library"""
+            options = []
+            if text == "":
+                options += [str(i)+":" for i in range(n_experiments)]
+            if ":" in text:
+                exp_id, text = text.split(":")
+                channels = channels_per_experiment.values()[int(exp_id)]
+                options += [
+                    exp_id+":"+ch for ch in channels if ch.startswith(text)
+                ]
+            else:
+                options += [
+                    ch for ch in all_channels if ch.startswith(text)
+                ]
+
+            if state >= len(options):
+                return None
+            return sorted(options)[state]
+
+        readline.parse_and_bind("tab: complete")
+        readline.set_completer_delims(" \t,;")
+        readline.set_completer(completer)
+
+        print()
+        print("Experiments loaded:")
+        for i, exp_name in enumerate(experiments.keys()):
+            print("    {}: {}".format(i, exp_name))
+        print(PROMPT_HEADER)
+        while 1:
+            # Note: input() uses python3 sematics (provided by six)
+            channel_spec = input("blocks-plot> ") 
+            if channel_spec in ["q", "quit", "exit", "e"]:
+                break
+
+            channel_spec = channel_spec.split(',')
+            matched = match_channel_specs(experiments, channel_spec)
+
+            if len(matched.columns) == 0:
+                print("Your specification did not match any channels.")
+                continue
+
+            plot_dataframe(matched)
+            print()
+
+       
+
+if __name__ == "__main__":
```

Comment:
When you install using pip or setup.py it will copy the files in bin/ in a
folder on the user's path, so you can't import from it anymore, so you
won't be able to share code.

I haven't looked at this PR yet, but I'm not 100% sure you'll need to put
the code in blocks/scripts. It depends on whether you think the code could
ever be imported by another script or function. If this script is
interactive, this might not be the case. On the other hand, if the plotting
logic could be useful for other scripts, then put it in blocks/scripts.
Hope that made sense.
On Feb 17, 2015 5:10 PM, "Jörg Bornschein" <notifications@github.com> wrote:

> In bin/blocks-plot
> <https://github.com/bartvm/blocks/pull/300#discussion_r24859536>:
>
> > +            if channel_spec in ["q", "quit", "exit", "e"]:
> > +                break
> > +
> > +            channel_spec = channel_spec.split(',')
> > +            matched = match_channel_specs(experiments, channel_spec)
> > +
> > +            if len(matched.columns) == 0:
> > +                print("Your specification did not match any channels.")
> > +                continue
> > +
> > +            plot_dataframe(matched)
> > +            print()
> > +
> > +
> > +
> > +if __name__ == "__main__":
>
> Done.
>
> Is it easier to write unit test if it's done like this? Or what's the
> rationale behind this pattern?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bartvm/blocks/pull/300/files#r24859536>.
>